packages/jazz-react-core/src/auth/MagicLinkAuth.tsx

```diff
@@ -0,0 +1,384 @@
+    | "confirmationCodeIncorrect"
+    | "authorized"
+    | "error"
+  >("idle");
+
+  const magicLinkAuth = useMemo(() => {
+    return new MagicLinkAuth(
+      context.node.crypto,
+      context.authenticate,
+      authSecretStorage,
+      origin,
+      options,
+    );
+  }, [origin, options]);
+
+  const createLink = useCallback(async () => {
+    let transfer = await magicLinkAuth.createTransferAsConsumer();
+
+    const url = magicLinkAuth.createLink("provider", transfer);
+
+    async function handleFlow() {
+      try {
+        // Wait for the provider to accept the transfer
+        setStatus("waitingForProvider");
+        transfer = await waitForCoValueCondition(
+          transfer,
+          {},
+          (t) => Boolean(t.acceptedBy),
+          handlerTimeout,
+        );
+
+        setStatus("confirmationCodeRequired");
+        const code = await new Promise<string>((resolve) => {
+          setSendConfirmationCode(() => (code: string) => resolve(code));
+        });
+        transfer.confirmationCodeInput = code;
+        setStatus("confirmationCodePending");
+
+        // Wait for provider to reject or confirm and reveal the secret
+        transfer = await waitForCoValueCondition(
+          transfer,
+          { resolve: {} },
+          (t) => t.status === "incorrectCode" || Boolean(t.secret),
+          handlerTimeout,
+        );
+
+        if (transfer.status === "incorrectCode") {
+          setStatus("confirmationCodeIncorrect");
+          return;
+        }
+
+        // Log in using the transfer secret
+        await magicLinkAuth.logInViaTransfer(transfer);
+        setStatus("authorized");
+        onLoggedIn?.();
+      } catch (error) {
+        console.error("Magic Link Auth error", error);
+        setStatus("error");
+      }
+    }
+
+    handleFlow();
+
+    return url;
+  }, [magicLinkAuth]);
+
+  return {
+    status,
+    createLink,
+    sendConfirmationCode,
+  } as const;
+}
+
+export function useHandleMagicLinkAuthAsConsumer(
+  origin: string,
+  url: string,
+  {
+    handlerTimeout = 30 * 1000,
+    onLoggedIn,
+    ...options
+  }: Partial<MagicLinkAuthConsumerOptions> = {},
+) {
+  const context = useJazzContext();
+  const authSecretStorage = useAuthSecretStorage();
+  const hasRunRef = useRef(false);
+  const [sendConfirmationCode, setSendConfirmationCode] = useState<
+    null | ((code: string) => void)
+  >(null);
+
+  const [status, setStatus] = useState<
+    | "idle"
+    | "confirmationCodeRequired"
+    | "confirmationCodePending"
+    | "confirmationCodeIncorrect"
+    | "authorized"
+    | "error"
+  >("idle");
+
+  const magicLinkAuth = useMemo(() => {
+    return new MagicLinkAuth(
+      context.node.crypto,
+      context.authenticate,
+      authSecretStorage,
+      origin,
+      options,
+    );
+  }, [origin, options]);
+
+  useEffect(() => {
+    if (hasRunRef.current) return;
+    hasRunRef.current = true;
+
+    async function handleFlow() {
```

Comment:
Does this flow handle cancelation?

What happens if the component gets unmounted or the user cancels the auth process?